readthedocs/proxito/views/serve.py

```diff
@@ -266,14 +264,17 @@ def get(self, request, proxito_path, template_name='404.html'):
         if default_version_slug != version_slug:
             versions.append(default_version_slug)
         for version_slug_404 in versions:
+            if not self.allowed_user(request, final_project, version_slug_404):
```

Comment:
Hmm. This is going to add even more queries to our slow 404's. I'm a bit worried about that. I guess it won't cause a query on the .org, which is where i'm most worried about it, so it should be fine.

Reply:
Yeah, on .org that function just returns True, it doesn't query the db.